misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobQueueModule.kt

```diff
@@ -102,38 +111,84 @@ class AwsSqsJobQueueModule(
       config.task_queue ?: RepeatedTaskQueueConfig(num_parallel_tasks = -1)
 
   private class AmazonSQSProvider(
-    val config: AwsSqsJobQueueConfig, val region: AwsRegion, val forSqsReceiving: Boolean
+    val config: AwsSqsJobQueueConfig,
+    val region: AwsRegion,
+    val forSqsReceiving: Boolean
   ) : Provider<AmazonSQS> {
     @Inject lateinit var credentials: AWSCredentialsProvider
+    @Inject lateinit var features: FeatureFlags
+    @Inject @AppName lateinit var appName: String
 
-    override fun get() = buildClient(config, credentials, region, forSqsReceiving)
+    override fun get() : AmazonSQS {
+      return if (forSqsReceiving) {
+        buildReceivingClient(credentials, region)
+      } else {
+        buildClient(appName, config, credentials, region, features)
+      }
+    }
   }
 
   companion object {
+    /** Build an unbuffered [AmazonSQS] client for receiving messages. */
+    private fun buildReceivingClient(
```

Comment:
I am tempted to convert this companion object into a Singleton factory whose constructor takes these arguments
```
class AmazonSQSFactory(
    @AppName appName: String,
    region: AwsRegion,
    credentials: AWSCredentialsProvider,
    features: FeatureFlags
) {
  fun create() ...
  fun createReceiving() ...
}
```
We can inject this in `provideSQSClient`, `provideSQSClientForReceiving`, and `AmazonSQSProvider`

This way we pass less arguments around

Reply:
Sounds like a good legibility improvement. I'll clean this up when we drop the feature-flag gated behavior.